lib/evaluators/AggregateEvaluator.ts

```diff
@@ -178,49 +182,45 @@ class Sum extends BaseAggregator<SumState> {
   }
 }
 
-type MinState = { minNum: number, minTerm: RDF.Term };
-class Min extends BaseAggregator<MinState> {
-  init(start: RDF.Term): MinState {
-    const { value } = extractNumericValueAndTypeOrError(start);
-    return { minNum: value, minTerm: start };
+type ExtremeState = { value: number, term: RDF.Term };
+class Min extends BaseAggregator<ExtremeState[]> {
```

Comment:
I didn't notice this before, but this change will make it so that this aggregator will keep _all_ passed terms in memory, which will lead to major performance problems, since the number of intermediary terms can be extremely high.

As such, I would keep the previous approach where only a singular state is maintained, instead of storing _all_ states.

Of course, this complicates the approach we wanted to take here (where string-based comparison is done when we have different types), so perhaps we need to reconsider that choice.

Perhaps we can just attempt a `parseInt` on non-numbers, and if we have a`NaN`, we simply ignore the value. What do you think?

(in this case, make sure to add a test where there are only string terms that result in `NaN`)